internal/client/cluster/sourcesecret/source_secret_resource.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package sourcesecretclient
+
+import (
+	"net/url"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/transport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	sourcesecret "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/sourcesecret/cluster"
+)
+
+const (
+	apiVersionAndGroup                 = "v1alpha1/clusters"
+	apiKind                            = "fluxcd/sourcesecrets"
+	namespaces                         = "namespaces"
+	queryParamKeyManagementClusterName = "fullName.managementClusterName"
+	queryParamKeyProvisionerName       = "fullName.provisionerName"
+)
+
+// New creates a new secret resource service API client.
+func New(transport *transport.Client) ClientService {
+	return &Client{Client: transport}
+}
+
+/*
+Client for secret resource service API.
+*/
+type Client struct {
+	*transport.Client
+}
+
+// ClientService is the interface for Client methods.
+type ClientService interface {
+	SourceSecretResourceServiceCreate(request *sourcesecret.VmwareTanzuManageV1alpha1ClusterFluxcdSourceSecretRequest) (*sourcesecret.VmwareTanzuManageV1alpha1ClusterFluxcdSourcesecretResponse, error)
```

Comment:
Let's rename all CRUD calls in this format: `VmwareTanzuManageV1alpha1ClusterFluxcdSourcesecretResourceService<Operation>`

Reply:
@Vasundharashukla , this is too long, we should consider reducing the lenght

Reply:
Then let's remove `VmwareTanzu` from the prefix, I just checked the cluster group resource where the name of the method was in the above format. Also, @Shubbhang351 remove the prefix from cluster group scope methods too.

Reply:
updated